bin/util.ml

```diff
@@ -47,6 +50,8 @@ let check_path contexts =
           match Path.Source.split_first_component src with
           | None -> internal_path ()
           | Some (ctx, src) ->
+            let ctx = Dune.Context_name.parse_string_exn (Loc.none, ctx) in
```

Comment:
I'm not sure raising an exn in here is very good. I suspect the user can reach this case by making a typo in a build path somewhere and dune will crash in the middle of the build.

Reply:
We can do it, but doesn't it make the error slightly worse? It's not the same thing to write an invalid context name vs writing the name of a context that doesn't exist.

Reply:
Actually, I just ended up reusing your validation code from `Dpath`. Have a look at the last commit.